vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ExternalLinkManager.java

```diff
@@ -98,7 +120,67 @@ protected void removeFromLinkList(ISpace space, IExternalLink link) {
 
     @Override
     protected void deleteLinkRepo(IExternalLink link) {
-        externalLinkRepo.delete((ExternalLink)link);
+        externalLinkRepo.delete((ExternalLink) link);
+    }
+
+    @Override
+    public IExternalLinkDisplay createLink(String title, String id, float positionX, float positionY, int rotation,
+            String linkedId, String linkLabel, DisplayType displayType, byte[] linkImage, String imageFilename,
+            String howToOpen)
+            throws SpaceDoesNotExistException, ImageCouldNotBeStoredException, SpaceDoesNotExistException {
+
+        IExternalLink link = createLinkObject(title, id);
+        ExternalLinkValue target = getTarget(linkedId);
+        link.setName(linkLabel);
+        link.setTarget(target);
+        IExternalLinkDisplay displayLink = createDisplayLink(link);
+        setDisplayProperties(displayLink, id, positionX, positionY, rotation, displayType, linkImage, imageFilename,
+                howToOpen);
+        return updateLinkAndDisplay(link, displayLink);
```

Comment:
the above code is the same as for `createLink` from the superclass, isn't it? If so, I would all the superclass method and then simply set the `howToOpen` property on the returned display object.